src/main/java/com/spotify/reaper/service/SegmentRunner.java

```diff
@@ -236,19 +240,35 @@ private void runRepair() {
     }
     LOG.debug("Exiting synchronized section with segment ID {}", segmentId);
   }
+  
+  private void declineRun() {
+    LOG.info("SegmentRunner declined to repair segment {} because only one segment is allowed "
+        + "at once for incremental repairs", segmentId);
+    String msg = "Postponed due to already running segment";
+    repairRunner.updateLastEvent(msg);
+  }
 
   boolean canRepair(RepairSegment segment, String keyspace, JmxProxy coordinator,
       LazyInitializer<Set<String>> busyHosts) {
     Collection<String> allHosts;
     if(repairUnit.getIncrementalRepair()){
     	// In incremental repairs, only one segment is allowed at once (one segment == the full primary range of one node)
     	if(repairHasSegmentRunning(segment.getRunId())) {
-    		LOG.info("SegmentRunner declined to repair segment {} because only one segment is allowed "
-                    + "at once for incremental repairs", segmentId);
-           String msg = "Postponed due to already running segment";
-           repairRunner.updateLastEvent(msg);
-           return false;
+    	  declineRun();
+        return false;
+    	}
+    	
+    	for(RepairSegment segmentInRun:context.storage.getRepairSegmentsForRun(segment.getRunId())){
+    	  try (JmxProxy hostProxy = context.jmxConnectionFactory.connect(segmentInRun.getCoordinatorHost())) {
+    	    if(hostProxy.isRepairRunning()) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)